docs: update admin api English doc (#8227)
apache/apisix: docs/en/latest/admin-api.md

#
-->

## Description

The Admin API lets users control their deployed Apache APISIX instance. The [architecture design](./architecture-design/apisix.md) gives an idea about how everything fits together.

## Configuration

When APISIX is started, the Admin API will listen on port `9180` by default and take the API prefixed with `/apisix/admin`.

> Therefore, to avoid conflicts between your designed API and `/apisix/admin`, you can modify the configuration file [`/conf/config.yaml`](https://github.com/apache/apisix/blob/master/ conf/config.yaml) to modify the default listening port.

Reviewer: ```suggestion
Therefore, to avoid conflicts between your designed API and `/apisix/admin`, you can modify the configuration file [`/conf/config.yaml`](https://github.com/apache/apisix/blob/master/conf/config.yaml) to modify the default listening port.
```

APISIX supports setting the IP access allowlist of Admin API to prevent APISIX from being illegally accessed and attacked. You can configure the IP addresses to allow access in the `deployment.admin.allow_admin` option in the `./conf/config.yaml` file.

The `X-API-KEY` shown below refers to the `deployment.admin.admin_key.key` in the `./conf/config.yaml` file, which is the access token for the Admin API.

:::tip

> It is recommended that you modify the default listening port, IP access allowlist and Admin API token of the Admin API to ensure the security of your API.

Reviewer: ```suggestion
For security reasons, please modify the default `admin_key`, and check the `allow_admin` IP access list.
```

| protocol.conf | False    | Configuration | Protocol-specific configuration.                             |                    |

To learn more about filtering in stream proxies, check [this](./stream-proxy.md#more-route-match-options) document.

## kms

kms means `Secrets Management`, which could use any secret manager supported, e.g. `vault`.

### kms API

> kms resource request address:: /apisix/admin/kms/{secretmanager}/{id}

Reviewer: ```suggestion
kms resource request address: /apisix/admin/kms/{secretmanager}/{id}
```

Reviewer: Also, should we add a ``` around API?

Author: Let's not modify it now, let us modify it together in the next version.
